spec/controllers/your_details_controller_spec.rb

```diff
@@ -1,61 +1,115 @@
 module Wpcc
   describe YourDetailsController do
     routes { Wpcc::Engine.routes }
+    let(:data_for_form) do
+      {
+        'age' => 34,
+        'salary' => 30_000,
+        'gender' => 'female',
+        'salary_frequency' => 'month',
+        'contribution_preference' => 'full'
+      }
+    end
 
     describe '#new' do
-      context 'english' do
+      subject { response }
+
+      context 'supported translations' do
         it 'renders the start view for english' do
           get :new, locale: 'en'
-
-          expect(response).to be_success
+          is_expected.to be_success
         end
-      end
 
-      context 'welsh' do
         it 'renders the start view for welsh' do
           get :new, locale: 'cy'
-
-          expect(response).to be_success
+          is_expected.to be_success
         end
       end
 
-      context 'translation not supported' do
-        it 'throws an error for an unsupported locale' do
+      context 'unsupported translation' do
+        it 'throws an error for french' do
           expect do
             get :new, locale: 'fr'
           end.to raise_error ActionController::UrlGenerationError
         end
       end
 
-      context 'when editing details which have previously been submitted' do
-        let(:your_details_form) { Wpcc::YourDetailsForm.new(session) }
-        let(:session) do
-          {
-            'age' => 34,
-            'gender' => 'female',
-            'salary' => 34_125,
-            'salary_frequency' => 'year',
-            'contribution_preference' => 'minimum'
-          }
-        end
-
+      context 'editing previously submitted form' do
         it 'instantiates a YourDetailsForm object with the session details' do
+          get :new, nil, data_for_form
           expect(Wpcc::YourDetailsForm)
             .to receive(:new)
-            .with(session)
-            .and_return(your_details_form)
+            .with(data_for_form)
 
-          get :new, nil, session
+          get :new
```

Comment:
```
- get :new, nil, session
+ get :new
```
why am I deleting these other details? because we don't want to pass **any** params to the controller. it should already have the data stored in it's session